controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -174,6 +183,71 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageConsumerReconciler) reconcileOnboardingSecret() error {
+	secret := &corev1.Secret{}
+	secret.Name = fmt.Sprintf("onboarding-token-%d", util.FnvHash(string(r.storageConsumer.UID)))
+	secret.Namespace = r.storageConsumer.Namespace
+	if err := r.get(secret); client.IgnoreNotFound(err) != nil {
+		return fmt.Errorf("failed to get secret %s: %v", secret.Name, err)
+	}
+
+	if !r.storageConsumer.Spec.Enable {
```

Comment:
What about the case where the secret exists but the key does not?

Reply:
that's an outside involvement and user needs to delete the secret to regenerate it.